Frontend/student-hub/src/components/Login.js

```diff
@@ -25,18 +27,59 @@ function Login() {
             // if correct navigate to the dashboard
             if (response.ok) {
                 const data = await response.json();
+                // check for userId in the data and not undefined
+                if (data.userId !== undefined){
+
+                  //store token and userId in sessionstorgae
                 sessionStorage.setItem('token', data.token);
-                navigate('/dashboard');
+                sessionStorage.setItem('userId', data.userId)
+                // Check if the userId exist in the database
+                const isValidUser = await checkUserId(data.userId);
+
+                if (isValidUser) {
+                  navigate('/dashboard');
+                  setLoading(false);
+                } else {
+                  setError('Invalid user credentials.');
+                  setOpen(true);
+                }
+              } else {
+                setError('Invalid server response. Please try again.');
+                setOpen(true);
+              }
               } else {
                 const { error } = await response.json();
                 setError(error);
+                setOpen(true);
               }
```

Comment:
there's a lot of nested if/else statements here, which make the logic a bit hard to trace. whenever you see this happening, i would try using a "return early" pattern here to make this cleaner!

so for example, you could start with:
```
if (!response.ok) {
  const { error } = await response.json();
  setError(error);
  setOpen(true);
  return;
}

// check if user ID is undefined
// check if user is valid
```

here's a blog post that describes the approach: https://gomakethings.com/the-early-return-pattern-in-javascript/